src/main/groovy/io/cify/runner/utils/CapabilityParser.groovy

```diff
@@ -68,25 +101,76 @@ class CapabilityParser {
-                    capabilitiesSet.add(capability)
+                    capability.getIos().add(ios)
+                    capability.getAndroid().add(android)
+                    capability.getBrowser().add(browser)
+                    capabilities.add(capability)
                 }
             }
         }
-        capabilitiesSet
+        capabilities
+    }
+
+    /**
+     * Gets capabilities from capabilities file content and puts each capability into a separate {@link Capabilities} object
+     *
+     * @param capabilitiesFileContent - capabilities file content
+     * @return the list of {@link Capabilities}
+     */
+    private static List getCapabilitiesOneByOne(LazyMap capabilitiesFileContent) {
+        List iOSCapabilitiesList = getCapabilitiesForType(capabilitiesFileContent, IOS)
+        List browserCapabilitiesList = getCapabilitiesForType(capabilitiesFileContent, BROWSER)
+        List androidCapabilitiesList = getCapabilitiesForType(capabilitiesFileContent, ANDROID)
+
+        List capabilities = []
+
+        iOSCapabilitiesList.each {
+            Capabilities capability = new Capabilities()
+            capability.getIos().add(it as LazyMap)
+            capabilities.add(capability)
+        }
+
+        androidCapabilitiesList.each {
+            Capabilities capability = new Capabilities()
+            capability.getIos().add(it as LazyMap)
+            capabilities.add(capability)
+        }
+
+        browserCapabilitiesList.each {
+            Capabilities capability = new Capabilities()
+            capability.getIos().add(it as LazyMap)
+            capabilities.add(capability)
+        }
+
+        capabilities
+    }
+
+    /**
+     * Gets capabilities from capabilities file content and puts them all to one {@link Capabilities} object
+     *
+     * @param capabilitiesFileContent - capabilities file content
+     * @return the list of {@link Capabilities}
+     */
+    private static List getCapabilitiesAllInOne(LazyMap capabilitiesFileContent) {
+        List capabilities = []
+        try {
+            capabilities.add(capabilitiesFileContent.get(CAPABILITIES) as Capabilities)
+            capabilities
+        } catch (all) {
+            LOG.debug(all.message, all)
+            throw new CifyPluginException("Could not read capabilities: " + all.message)
+        }
     }
 
     /**
      * Gets capability for device type
      *
      * @param capabilityFileContent - Capabilities json file content
      * @type - device type
-     * */
-    static List getCapabilitiesForType(LazyMap capabilityFileContent, String type) {
-        if (capabilityFileContent.get(SET) != null && capabilityFileContent.get(SET)[type] != null) {
-            return capabilityFileContent.get(SET)[type] as List
+     */
+    private static List getCapabilitiesForType(LazyMap capabilityFileContent, String type) {
+        if (capabilityFileContent.get(CAPABILITIES) != null && capabilityFileContent.get(CAPABILITIES)[type] != null) {
```

Comment:
I think capabilityFileContent.get(CAPABILITIES) can be stored to a variable and used then. It will make code reading easier and code itself a bit lighter (and shorter)

Reply:
Yep, done